### ISSUE TITLE
Fix hard-coding of CAS auth related URLS

### ISSUE THREAD

**Issue by kljensen** [kljensen/yale-class-chat#3]
https://github.com/kljensen/yale-class-chat/blob/4f99a597cbb503158e0a863f62d0aceb1001169d/app/config/config.exs#L30

This was quick and dirty. It needs to be passed in through the environment, per usual.

**Comment by nperanzi** [comment]
Does it make sense to turn base_url and callback into environment variables as well?
If so, is it better to store the entire URL in one variable or build the URL from multiple environment variables? My thought process for each option is below.

**Option 1**: URLs in single variables
`source.env` contains:
```
PORT=9001
MOCKCAS_PORT=9002
CAS_BASE_URL=http://localhost:9002/cas
CAS_CALLBACK=http://localhost:9001/auth/cas/callback
```

`config.exs` contains:
```
    base_url: System.get_env("CAS_BASE_URL"),
    callback: System.get_env("CAS_CALLBACK"),
```

**Option 2**:  Building URLs from multiple variables
`source.env` contains:
```
PORT=9001
MOCKCAS_PORT=9002
BASE_URL=http://localhost
```

`config.exs` contains:
```
    base_url: System.get_env("BASE_URL") <> ":" <> System.get_env("MOCKCAS_PORT") <> "/cas",
    callback: System.get_env("BASE_URL") <> ":" <> System.get_env("PORT") <> "/auth/cas/callback",
```

**Comment by kljensen** [comment]
Yes, all of them should be environment variables. Likely

```
CAS_BASE_URL
CAS_SERVICE_VALIDATE_BASE_URL
CAS_CALLBACK_URL
```

The reason for separate `CAS_SERVICE_VALIDATE_BASE_URL` in development is that the user (you) will get redirected to `CAS_BASE_URL` for CAS authentication. Once CAS redirects you back to the app, the elixir/phoenix app will contact the CAS server to verify your service ticket. Of course, the something like `http://localhost:XYZ` is not going to be accessible from inside in the docker container in which the app is running. So, we give it `http://mockcas:XYZ` in dev. Docker changes `/etc/hosts` so that you can refer to other services like that: `mockcas` resolves to the IP address of the mockcas service as defined in the `docker-compose.yaml`

**Comment by kljensen** [comment]
And, as an aside - there should generally be only ONE file that reads from the environment. One should not have `System.get_env` randomly in some code somewhere. That way, as dev, you know where to look for inputs that can control the system.

**Comment by kljensen** [comment]
Fixed by 48da8f0c96bc0a309fce7b5557a7483a065a171f
